File: Java/BlueJfiles/LibrarySystem/LibraryMember.java

/**
 * Write a description of class LibraryMember here.
 *
 * @author (your name)
 * @version (a version number or a date)
 */

import java.util.ArrayList;
public class LibraryMember
{
    // instance variables - replace the example below with your own
    private String name;
    private String user_id;
    private ArrayList <Book> book;
    private String messages;

    /**
     * Constructor for objects of class LibraryMember
     */
    
    public LibraryMember(String messages)
    { // initialise instance variables
        name = "";
        user_id = "";
        book = new ArrayList<Book>();
        this.messages = messages;   
    }
    
    public String get_name()
    {
        return name;
    }
    
    public String get_user_id()
    {
        return user_id;
    }
    
    public String get_messages()
    {
        return messages;
    }
    
    public void set_name(String new_name)
    {
        name = new_name;
    }
    
    public void set_user_id(String new_user_id)
    {
        user_id = new_user_id;
    }
    
    public void set_messages(String new_messages)
    {
        messages += new_messages;
    }
    
    public void add_book(Book new_book)
    {
        book.add(new_book);
    }
    
    public void print_books()
    {
        for (Book b : book)
        {
            System.out.println(b.print_details());
        }
    }
    
    public void get_number_borrowed()
    {
        System.out.println(book.size());
    }
}

File: python/LibrarySystems/library_member.py
class LibraryMember:
    def __init__(self, name, user_id, messages):
        self.name = name
        self.user_id = user_id
        self.borrowed = []
        self.messages = messages
    # This class represents a library member

    def get_name(self):
        return self.name
    # Returns the library member name

    def set_name(self, name):
        self.name = name
    # Sets name to given value

    def get_user_id(self):
        return self.user_id
    # Returns the library member's user id

    def set_user_id(self, user_id):
        self.user_id = user_id
    # Sets name to user id value

    def get_borrowed(self):
        return self.borrowed
    # Returns the books borrowed by library member

    def set_borrowed(self, borrowed):
        self.borrowed = borrowed
    # Sets books borrowed

    def get_messages(self):
        return self.messages
    # Returns the messages received

    def set_messages(self, messages):
        self.messages += messages
    # Sets messages received

    def print_user_details(self):
        print("The library member name: " + str(self.name))
        print("The library user ID is: " + str(self.user_id))
        print("The library member book(s) borrowed: " + str(self.borrowed))
        print("Messages received: " + str(self.messages))
    # Prints the details of library member

    def print_messages(self):
        print(self.messages)
    # Prints all the messages received by library member

    def add_book(self, book):
        self.borrowed.append(book)
    # Adds a borrowed book the list of borrowed books for the library member

    def print_books(self):
        for book in self.borrowed:
            book.print_book_details()
    # Prints all the details of books currently borrowed

    def get_number_borrowed(self):
        return len(self.borrowed)
    # Returns the number of borrowed books by a library member







